Debian #455484
GNOME #482159
Security issue: leak of the clipboard contents when the screen is 
locked.
--- trunk/src/gs-lock-plug.c	2007/11/09 21:44:42	1323
+++ trunk/src/gs-lock-plug.c	2007/12/12 21:15:16	1337
@@ -1563,6 +1563,19 @@
         return TRUE; /* Do not destroy */
 }
 
+static void
+clear_clipboards (GSLockPlug *plug)
+{
+        GtkClipboard *clipboard;
+
+        clipboard = gtk_widget_get_clipboard (GTK_WIDGET (plug), GDK_SELECTION_PRIMARY);
+        gtk_clipboard_clear (clipboard);
+        gtk_clipboard_set_text (clipboard, "", -1);
+        clipboard = gtk_widget_get_clipboard (GTK_WIDGET (plug), GDK_SELECTION_CLIPBOARD);
+        gtk_clipboard_clear (clipboard);
+        gtk_clipboard_set_text (clipboard, "", -1);
+}
+
 #define INVISIBLE_CHAR_DEFAULT       '*'
 #define INVISIBLE_CHAR_BLACK_CIRCLE  0x25cf
 #define INVISIBLE_CHAR_WHITE_BULLET  0x25e6
@@ -1578,6 +1591,8 @@
 
         plug->priv = GS_LOCK_PLUG_GET_PRIVATE (plug);
 
+        clear_clipboards (plug);
+
 #ifdef WITH_LIBNOTIFY
         plug->priv->leave_note_enabled = TRUE;
 #else
